### ISSUE TITLE
Add TLS configuration

### ISSUE THREAD

**Issue by tjcelaya** [autopilotpattern/consul#49]
As a result of the current implementation of #48, RPC traffic is exposed publicly and should therefore be encrypted. Since consul provides [mechanisms to do so](https://www.consul.io/docs/agent/encryption.html#rpc-encryption-with-tls) we should include a way to inject certs into the containers before consul can start in a similar fashion to how autopilotpattern/vault uses `docker exec` to bootstrap.

The proposed design is to check for `CONSUL_TLS_PATH` during preStart and if present, wait for a file to appear the the specified path. Gossip key configuration can be done by specifying `CONSUL_ENCRYPT_PATH` or `CONSUL_ENCRYPT_BASE64`.

**Comment by misterbisson** [comment]
This is implemented in https://github.com/autopilotpattern/vault/

**Comment by tjcelaya** [comment]
Right, the plan was to adapt the strategy describe in [this section](https://github.com/autopilotpattern/vault/#encrypted-communications) of the autopilotpattern/vault README to get the key material into the containers and use `CONSUL_TLS_PATH` to signal to ContainerPilot where the key will appear. Leaving that environment variable empty would indicate no TLS was being configured.
